orderbook/src/database/orders.rs

```diff
@@ -298,15 +303,39 @@ mod tests {
             .await
             .unwrap();
         assert_eq!(db_orders[0].order_meta_data.invalidated, false);
-        db.cancel_order(&order.order_meta_data.uid).await.unwrap();
+
+        let cancellation_time =
+            DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(1234567890, 0), Utc);
+        db.cancel_order(&order.order_meta_data.uid, cancellation_time)
+            .await
+            .unwrap();
         let db_orders = db
             .orders(&filter)
             .try_collect::<Vec<Order>>()
             .await
             .unwrap();
         assert_eq!(db_orders[0].order_meta_data.invalidated, true);
-        // TODO - cancel twice and verify that first cancellation date isn't over written
-        // This will require querying the DB for the cancellation_date.
+
+        let query = "SELECT cancellation_timestamp FROM orders;";
+        let first_cancellation: CancellationQueryRow =
+            sqlx::query_as(query).fetch_one(&db.pool).await.unwrap();
+        assert_eq!(cancellation_time, first_cancellation.cancellation_timestamp);
+
+        // Cancel again and verify that cancellation timestamp was not changed.
+        let irrelevant_time =
+            DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(1234567890, 1), Utc);
```

Comment:
nit: Cutting it tight with 1ns difference (I'm not sure all postgres configs support this granularity). Maybe you could use a time that's a bit more different?

Reply:
That is a good point. I'd say increment the seconds by 1 to be sure.